### PR TITLE
nixos/bacula: adding types to all options

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Motivation for this change

fixes in the list of #76184

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/backup/bacula.nix [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # TODO: test configuration when building nixexpr (use -t parameter)
 # TODO: support sqlite3 (it's deprecate?) and mysql
 
-with lib;
+with lib; with types;
```
Comment (SuperSandro2000) [inline nixos/modules/services/backup/bacula.nix:6]: Please add types before the types.

**Comment by SuperSandro2000** [comment]
@Scriptkiddi please fix the eval error.

**Comment by Scriptkiddi** [comment]
@SuperSandro2000 done

**Comment by SuperSandro2000** [comment]
@Scriptkiddi please fix the merge conflict.

**Comment by SuperSandro2000** [comment]
Please do not merge master into a PR to solve a merge conflict. It is always required to use git rebase or otherwise the git history will be very confusing.

**Comment by Scriptkiddi** [comment]
@SuperSandro2000  forgot that github does that its fixed now

**Pull request merged**